great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -188,28 +198,28 @@ class SparkDFExecutionEngine(ExecutionEngine):
     def __init__(
         self,
         *args,
-        persist=True,
-        spark_config=None,
-        force_reuse_spark_context=True,
+        persist: bool = True,
+        spark_config: Optional[dict] = None,
+        force_reuse_spark_context: Optional[bool] = None,
         **kwargs,
     ) -> None:
         self._persist = persist
 
-        if spark_config is None:
-            spark_config = {}
-
-        spark: pyspark.SparkSession = get_or_create_spark_application(
+        self.spark: pyspark.SparkSession = get_or_create_spark_session(
             spark_config=spark_config,
-            force_reuse_spark_context=force_reuse_spark_context,
         )
 
-        spark_config.update({k: v for (k, v) in spark.sparkContext.getConf().getAll()})
-
-        self.spark = spark
-
         azure_options: dict = kwargs.pop("azure_options", {})
         self._azure_options = azure_options
 
+        if force_reuse_spark_context is not None:
+            warnings.warn(
+                "force_reuse_spark_context is deprecated and will be removed in version 1.0. "
+                "In environments that allow it, the existing Spark context will be reused, adding the "
+                "spark_config options that have been passed. If the Spark context cannot be updated with "
+                "the spark_config, the context will be stopped and restarted with the new spark_config.",
+                category=RuntimeWarning,
```

Comment:
```suggestion
                category=DeprectationWarning,
```
https://docs.python.org/3/library/exceptions.html#DeprecationWarning